apps/guardian-ui/src/admin/FederationAdmin.tsx

```diff
@@ -17,13 +17,48 @@ import {
 import { CopyInput } from '@fedimint/ui';
 import { useTranslation } from '@fedimint/utils';
 import { useAdminContext } from '../hooks';
-import { ConfigResponse, Gateway, StatusResponse, Versions } from '../types';
+import {
+  ClientConfig,
+  ConfigResponse,
+  FederationStatus,
+  Gateway,
+  PeerStatus,
+  StatusResponse,
+  Versions,
+} from '../types';
 import { AdminMain } from '../components/AdminMain';
 import { ConnectedNodes } from '../components/ConnectedNodes';
 import { LightningModuleRpc } from '../GuardianApi';
 import { ReactComponent as CopyIcon } from '../assets/svgs/copy.svg';
 import { Pill } from '../components/Pill';
 
+interface PeerData {
+  id: number;
+  name: string;
+  status: PeerStatus;
+}
+
+function generatePeerData(
+  config: ClientConfig,
+  federationStatus: FederationStatus
+): PeerData[] {
+  const peerDataArray: PeerData[] = [];
+  for (const [id, status] of Object.entries(federationStatus.status_by_peer)) {
+    const numericId = parseInt(id, 10);
+    if (Object.prototype.hasOwnProperty.call(config.api_endpoints, numericId)) {
```

Comment:
can we skip the outer `hasOwnProperty` check? It doesn't seem very useful if endpoint is still nullable and we can just go with 
```ts
const endpoint = config.api_endpoints[numericId];
      if (endpoint) {
        peerDataArray.push({
          id: numericId,
          name: endpoint.name,
          status,
        });
      }
      ```